File: scripts/abilities.lua
function ability_roll()
  local total = -6
  local min = 6
  for i = 1, 4 do
    local roll = math.random(1, 6)
    if (roll < min) then
      total = total + min
      min = roll
    else
      total = total + roll
    end
  end
  return total
end

math.randomseed(os.time())
local scores = {}
for i = 1, 6 do
  table.insert(scores, ability_roll())
end
print(table.concat(scores, ' '))

File: scripts/abilities.py
import random

def ability_roll():
    total = -6
    min_roll = 6
    for _ in xrange(4):
        roll = random.randint(1, 6)
        if (roll < min_roll):
            total += min_roll
            min_roll = roll
        else:
            total += roll
    return total

scores = []
for _ in xrange(6):
    scores.append(ability_roll())
print(" ".join(str(x) for x in scores))
print("Sum: %s" % sum(scores))
